src/org.xtuml.bp.ui.marking/src/org/xtuml/bp/ui/marking/MarkingData.java

```diff
@@ -149,6 +161,11 @@ private void populateMarkings() {
 	 * @return String - An error message if issues were found, otherwise empty string
 	 */
 	public String validateFeatures() {
+		// doesn't hurt to populate here, incase the features file was added
+		// after the project creation
+		if(featureMap.isEmpty()) {
+			populateFeatures();
+		}

```

Comment:
What would happen if I edit "features.mark" (e.g. via config management) and refresh the project (assuming I already had some features loaded in the map)? Do I need to close and re-open BridgePoint?

Reply:
Right now if you were to make changes to the features.mark file and refresh nothing would happen.  You would not get those new changes and your would not lose the in-memory changes.  You would lose in-memory changes if you were to close the project and reopen it.  This issue does not attempt to address a marking file listener and merge in-memory changes with those on disk.  Another issue should be raised if such an extension of the marking functionality is desired.